style-guides/javascript/README.md

```diff
@@ -0,0 +1,1479 @@
+    + Use the literal syntax for object creation.
+    
+    ```javascript
+    // bad
+    var item = new Object();
+
+    // good
+    var item = {};
+    ```
+  - **Arrays**
+    + Use myArray.push(x) instead of myArray[myArray.length] = x
+    + Use Array#slice to copy an array
+  
+  - **Strings**
+    + **Note: Thoughts on always single quotes vs double quotes? Other say single quotes**
+
+  - **Functions**
+
+  - **Properties**
+    + Access properties via dot notation (myObj.name) rather than subscript notation (myObj[name])
+
+  - **Variables**
+    + Always use `var` to declare variables
+    + Use one `var` declaration per variable
+    + Assign variables at the top of their scope
+    + Declare unassigned variables last
+    + Declare one variable per line **Note: thoughts on this?**
+    + Use names that make sense. Don't use reserved words. Avoid single letters
+
+  - **Hoisting** [Read this section. It's good.](#hoisting)
+
+  - **Conditional Expressions & Equality**
+    + Use `===` and `!==` over `==` and `!=`
+    + Avoid duplication like `if (name !== '')`. Use `if (name)` instead.
+
+  - **Blocks**
+    + Separate into multiple lines and use braces
+      
+      ```javascript
+      // bad
+      if (true) { return true; }
+      
+      // good
+      if (true) {
+        return true;
+      }
+      ```
+
+  - **Comments**
+    + **NOTE: do people prefer `//` or `/** ... */` for multiline comments?**
+    + Comments should go over the line they refer to rather than to the right on the same line
+
+  - **Spacing Note: thoughts on all of this**
+    + **Note: thoughts on tabs vs spaces for indentation???**
+    + Place 1 space before the leading brace. Yes: `function() {`. No: `function(){`
+    + Focus on external spacing rather than internal spacing and follow conventional English.
+
+    ```javascript
+      //bad
+      for ( var i = 0 ; i < 3 ; i++ )
+
+      //good
+      for (var i = 0; i < 3; i++)
+
+      //bad
+      [apples,bananas,cherries]
+      [ apples , bananas , cherries ]
+
+      //good
+      [apples, bananas, cherries]
+    ```
+    + Yes: `if (blah) {`. No: `if(blah){`
+    + Any `,` and `;` must not have preceding space
+
+  - [**Resources**](#resources) 
+
+
+  
+## Types
+
+  - **Primitives**: When you access a primitive type you work directly on its value
+
+    + `string`
+    + `number`
+    + `boolean`
+    + `null`
+    + `undefined`
+
+    ```javascript
+    var foo = 1;
+    var bar = foo;
+
+    bar = 9;
+
+    console.log(foo, bar); // => 1, 9
+    ```
+  - **Complex**: When you access a complex type you work on a reference to its value
+
+    + `object`
+    + `array`
+    + `function`
+
+    ```javascript
+    var foo = [1, 2];
+    var bar = foo;
+
+    bar[0] = 9;
+
+    console.log(foo[0], bar[0]); // => 9, 9
+    ```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Objects
+
+  - Use the literal syntax for object creation.
+
+    ```javascript
+    // bad
+    var item = new Object();
```

Comment:
There are somethings that new Object lets you do around setting immutable keys...